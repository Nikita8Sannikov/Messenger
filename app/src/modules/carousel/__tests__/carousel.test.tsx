import { render, screen } from "@testing-library/react";

import Carousel, { type ICarouselImage } from "@modules/carousel";

describe("Carousel Component", () => {
	const mockImage: ICarouselImage = {
		src: "test-image.jpg",
		alt: "Test Image",
		authorName: "John Doe",
		dateTime: "2024-03-20",
		authorAvatarUrl: "avatar.jpg",
	};

	it("renders carousel with single image", () => {
		render(<Carousel image={mockImage} activeKey={0} allCount={1} />);
        
		expect(screen.getByTestId("carousel")).toBeInTheDocument();
		expect(screen.getByAltText("Test Image")).toBeInTheDocument();
		expect(screen.queryByRole("button")).not.toBeInTheDocument();
	});

	it("renders carousel with multiple images and navigation", () => {
		render(<Carousel image={mockImage} activeKey={0} allCount={3} />);
        
		expect(screen.getByTestId("carousel")).toBeInTheDocument();
		expect(screen.getByAltText("Test Image")).toBeInTheDocument();
        
		const buttons = screen.getAllByRole("button");
		expect(buttons).toHaveLength(2);
        
		expect(buttons[0]).toBeDisabled();
		expect(buttons[1]).not.toBeDisabled();
	});

	it("disables next button on last image", () => {
		render(<Carousel image={mockImage} activeKey={2} allCount={3} />);
        
		const buttons = screen.getAllByRole("button");
		expect(buttons[0]).not.toBeDisabled();
		expect(buttons[1]).toBeDisabled();
	});

	it("enables both buttons on middle image", () => {
		render(<Carousel image={mockImage} activeKey={1} allCount={3} />);
        
		const buttons = screen.getAllByRole("button");
		expect(buttons[0]).not.toBeDisabled();
		expect(buttons[1]).not.toBeDisabled();
	});

	it("displays correct image properties", () => {
		render(<Carousel image={mockImage} activeKey={0} allCount={1} />);
        
		const image = screen.getByAltText("Test Image");
		expect(image).toHaveAttribute("src", "test-image.jpg");
	});
});

describe("Carousel Component Snapshot Tests", () => {
	const mockImage: ICarouselImage = {
		src: "test-image.jpg",
		alt: "Test Image",
		authorName: "John Doe",
		dateTime: "2024-03-20",
		authorAvatarUrl: "avatar.jpg",
	};

	it("should match snapshot with single image", () => {
		const { container } = render(
			<Carousel image={mockImage} activeKey={0} allCount={1} />,
		);
		expect(container).toMatchSnapshot();
	});

	it("should match snapshot with multiple images - first image", () => {
		const { container } = render(
			<Carousel image={mockImage} activeKey={0} allCount={3} />,
		);
		expect(container).toMatchSnapshot();
	});

	it("should match snapshot with multiple images - middle image", () => {
		const { container } = render(
			<Carousel image={mockImage} activeKey={1} allCount={3} />,
		);
		expect(container).toMatchSnapshot();
	});

	it("should match snapshot with multiple images - last image", () => {
		const { container } = render(
			<Carousel image={mockImage} activeKey={2} allCount={3} />,
		);
		expect(container).toMatchSnapshot();
	});

	it("should match snapshot with different image properties", () => {
		const newImage = {
			...mockImage,
			src: "new-image.jpg",
			alt: "New Test Image",
		};
		const { container } = render(
			<Carousel image={newImage} activeKey={0} allCount={1} />,
		);
		expect(container).toMatchSnapshot();
	});

	it("should match snapshot with large number of images", () => {
		const { container } = render(
			<Carousel image={mockImage} activeKey={5} allCount={10} />,
		);
		expect(container).toMatchSnapshot();
	});
});
