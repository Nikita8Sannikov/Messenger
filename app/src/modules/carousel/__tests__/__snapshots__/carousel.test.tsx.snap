// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel Component Snapshot Tests should match snapshot with different image properties 1`] = `
<div>
  <div
    class="carousel"
    data-testid="carousel"
  >
    <div
      class="carousel__photo"
    >
      <div
        class="photo"
      >
        <img
          alt="New Test Image"
          class="photo__image"
          data-testid="image"
          loading="eager"
          src="new-image.jpg"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Carousel Component Snapshot Tests should match snapshot with large number of images 1`] = `
<div>
  <div
    class="carousel"
    data-testid="carousel"
  >
    <div
      class="carousel__photo"
    >
      <div
        class="photo"
      >
        <img
          alt="Test Image"
          class="photo__image"
          data-testid="image"
          loading="eager"
          src="test-image.jpg"
        />
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiMobileStepper-root MuiMobileStepper-positionStatic carousel__photo__stepper css-14xii3f-MuiPaper-root-MuiMobileStepper-root"
        style="--Paper-shadow: none;"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary css-1ccsd5i-MuiButtonBase-root-MuiButton-root"
          id=":rc:"
          tabindex="0"
          type="button"
        >
          <svg
            data-testid="arrow-left-icon"
            height="24px"
            viewBox="0 0 24 24"
            width="24px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.41 16.59 10.83 12l4.58-4.59L14 6l-6 6 6 6z"
              fill="currentColor"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="0"
            />
          </svg>
           
          Back
        </button>
        6
         / 
        10
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary css-1ccsd5i-MuiButtonBase-root-MuiButton-root"
          id=":rd:"
          tabindex="0"
          type="button"
        >
          Further
           
          <svg
            data-testid="arrow-right-icon"
            height="24px"
            viewBox="0 0 24 24"
            width="24px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8.59 16.59 13.17 12 8.59 7.41 10 6l6 6-6 6z"
              fill="currentColor"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="0"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Carousel Component Snapshot Tests should match snapshot with multiple images - first image 1`] = `
<div>
  <div
    class="carousel"
    data-testid="carousel"
  >
    <div
      class="carousel__photo"
    >
      <div
        class="photo"
      >
        <img
          alt="Test Image"
          class="photo__image"
          data-testid="image"
          loading="eager"
          src="test-image.jpg"
        />
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiMobileStepper-root MuiMobileStepper-positionStatic carousel__photo__stepper css-14xii3f-MuiPaper-root-MuiMobileStepper-root"
        style="--Paper-shadow: none;"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary Mui-disabled MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary css-1ccsd5i-MuiButtonBase-root-MuiButton-root"
          disabled=""
          id=":r6:"
          tabindex="-1"
          type="button"
        >
          <svg
            data-testid="arrow-left-icon"
            height="24px"
            viewBox="0 0 24 24"
            width="24px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.41 16.59 10.83 12l4.58-4.59L14 6l-6 6 6 6z"
              fill="currentColor"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="0"
            />
          </svg>
           
          Back
        </button>
        1
         / 
        3
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary css-1ccsd5i-MuiButtonBase-root-MuiButton-root"
          id=":r7:"
          tabindex="0"
          type="button"
        >
          Further
           
          <svg
            data-testid="arrow-right-icon"
            height="24px"
            viewBox="0 0 24 24"
            width="24px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8.59 16.59 13.17 12 8.59 7.41 10 6l6 6-6 6z"
              fill="currentColor"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="0"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Carousel Component Snapshot Tests should match snapshot with multiple images - last image 1`] = `
<div>
  <div
    class="carousel"
    data-testid="carousel"
  >
    <div
      class="carousel__photo"
    >
      <div
        class="photo"
      >
        <img
          alt="Test Image"
          class="photo__image"
          data-testid="image"
          loading="eager"
          src="test-image.jpg"
        />
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiMobileStepper-root MuiMobileStepper-positionStatic carousel__photo__stepper css-14xii3f-MuiPaper-root-MuiMobileStepper-root"
        style="--Paper-shadow: none;"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary css-1ccsd5i-MuiButtonBase-root-MuiButton-root"
          id=":ra:"
          tabindex="0"
          type="button"
        >
          <svg
            data-testid="arrow-left-icon"
            height="24px"
            viewBox="0 0 24 24"
            width="24px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.41 16.59 10.83 12l4.58-4.59L14 6l-6 6 6 6z"
              fill="currentColor"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="0"
            />
          </svg>
           
          Back
        </button>
        3
         / 
        3
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary Mui-disabled MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary css-1ccsd5i-MuiButtonBase-root-MuiButton-root"
          disabled=""
          id=":rb:"
          tabindex="-1"
          type="button"
        >
          Further
           
          <svg
            data-testid="arrow-right-icon"
            height="24px"
            viewBox="0 0 24 24"
            width="24px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8.59 16.59 13.17 12 8.59 7.41 10 6l6 6-6 6z"
              fill="currentColor"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="0"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Carousel Component Snapshot Tests should match snapshot with multiple images - middle image 1`] = `
<div>
  <div
    class="carousel"
    data-testid="carousel"
  >
    <div
      class="carousel__photo"
    >
      <div
        class="photo"
      >
        <img
          alt="Test Image"
          class="photo__image"
          data-testid="image"
          loading="eager"
          src="test-image.jpg"
        />
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiMobileStepper-root MuiMobileStepper-positionStatic carousel__photo__stepper css-14xii3f-MuiPaper-root-MuiMobileStepper-root"
        style="--Paper-shadow: none;"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary css-1ccsd5i-MuiButtonBase-root-MuiButton-root"
          id=":r8:"
          tabindex="0"
          type="button"
        >
          <svg
            data-testid="arrow-left-icon"
            height="24px"
            viewBox="0 0 24 24"
            width="24px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.41 16.59 10.83 12l4.58-4.59L14 6l-6 6 6 6z"
              fill="currentColor"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="0"
            />
          </svg>
           
          Back
        </button>
        2
         / 
        3
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary css-1ccsd5i-MuiButtonBase-root-MuiButton-root"
          id=":r9:"
          tabindex="0"
          type="button"
        >
          Further
           
          <svg
            data-testid="arrow-right-icon"
            height="24px"
            viewBox="0 0 24 24"
            width="24px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8.59 16.59 13.17 12 8.59 7.41 10 6l6 6-6 6z"
              fill="currentColor"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="0"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Carousel Component Snapshot Tests should match snapshot with single image 1`] = `
<div>
  <div
    class="carousel"
    data-testid="carousel"
  >
    <div
      class="carousel__photo"
    >
      <div
        class="photo"
      >
        <img
          alt="Test Image"
          class="photo__image"
          data-testid="image"
          loading="eager"
          src="test-image.jpg"
        />
      </div>
    </div>
  </div>
</div>
`;
